_posts/2023-02-01-user-defined-operators.md

```diff
@@ -0,0 +1,173 @@
+All parameters must be of the same type and this must be the type specified in `using for`.
+The return value must be of the same type as well, except for comparison operators, where it must be `bool`.
+
+A few points of note:
+- Private library functions are allowed but visible only within the library.
+- Built-in functions like `keccak256`, `revert` or `abi.encode()` cannot be used.
+- Library functions are allowed but contract and interface functions are not.
+    This means that the definition cannot be virtual.
+- Overloaded functions are not purposefully disallowed, but due to the lack of a language mechanism
+    for selecting a specific overload, it is currently not possible to use such functions.
+- Operator definition may be a public library function.
+    External functions will eventually be allowed too
+    (see [issue #13765](https://github.com/ethereum/solidity/issues/13765)).
+- Being a `pure` function, the operator defintion is allowed to make `pure` external calls.
+
+The rules for `using for global` with operators are the same as for bound functions:
+it is only allowed at file level, the type must be defined in the current source unit and the
+effects extend to wherever the type is visible, not just the current source unit.
+
+#### Can Operators Be Redefined?
+It is not possible to redefine any built-in operator.
+All valid cases are the ones where previously there was no built-in operator.
+
+Only one definition is allowed for each user-defined operators for a given type.
+Note that:
+- Unary and binary `-` count as two separate operators.
+- As long as the definition stays the same, it is not an error to repeat the `using for` directive,
+    just as it is not an error to attach the same function to a type twice.
+
+`using for` is not inherited by derived contracts and this applies to operators as well.
+For this reason, it is possible for a base contract and a derived contract to use different definitions
+for the same operator.
+The same is not possible at file level.
+If an operator is defined at file level, it cannot have a different definition inside a contract where
+that file-level definition is visible.
+
+### AST Changes
+The new feature introduces two new attributes to the AST:
+1. Functions listed in `functionList` on `UsingForDirective` nodes, may now optionally have an
+    `operator` attribute.
+    When present, it indicates that the function defines an operator and specifies its name.
+1. `BinaryOperation` and `UnaryOperation` may now have a `function` attribute.
+    If present, the attribute indicates that the operation is actually an invocation of a user-defined
+    operator and specifies the ID of the function used to define it.
+
+This effectively introduces a new way to perform a function call, not represented by a `FunctionCall` node.
+This will affect any tool that needs to detect and follow function calls, for example to build a
+control flow graph.
```

Comment:
```suggestion
### AST Changes
The new feature introduces two new attributes to the Abstract Syntax Tree (AST):
1. Functions listed in `functionList` on `UsingForDirective` nodes, may now optionally have an
    `operator` attribute.
    When present, it indicates that the function defines an operator and specifies its name.
1. `BinaryOperation` and `UnaryOperation` may now have a `function` attribute.
    If present, the attribute indicates that the operation is actually an invocation of a user-defined
    operator and specifies the ID of the function used to define it.

This effectively introduces a new way to perform a function call, not represented by a `FunctionCall` node.
This will affect any tool that needs to detect and follow function calls, for example, to build a
control flow graph.
```